components/micro-gateway-core/src/main/ballerina/src/gateway/filters/pre_authn_filter.bal

```diff
@@ -24,6 +24,13 @@ public type PreAuthnFilter object {
 
     public function filterRequest(http:Caller caller, http:Request request,@tainted http:FilterContext context) returns boolean {
         setFilterSkipToFilterContext(context);
+        boolean preserveUserAgentHeader = getConfigBooleanValue(SERVER_PRESERVE_HEADERS_ID,
```

Comment:
Lets write two  separete methods to preserver headers on request out and response out

Reply:
Done with https://github.com/wso2/product-microgateway/pull/1577/commits/66fd20ad3a8c7a58085ed98286a4e29d1663f8ce